XpringKit/PayID/PayIDClient.swift

```diff
@@ -29,20 +33,51 @@ public class PayIDClient {
   ///   - ach
   ///
   //  TODO: Link a canonical list at payid.org when available.
-  public init(network: String) {
+  public init(network: String, completionQueue: DispatchQueue = DispatchQueue.main) {
     self.network = network
   }
 
   /// Resolve the given PayID to an address.
   ///
   /// - Parameter payID: The PayID to resolve for an address.
-  /// - Parameter completion: A closure called with the result of the operation.
   /// - Returns: An address representing the given PayID.
-  // TODO(keefertaylor): Make this API synchronous to mirror functionality provided by ILP / XRP.
+  public func address(for payID: String) throws -> Result<CryptoAddressDetails, PayIDError> {
+    // Assign a bogus value. This will get overwritten when the asynchronous call is made.
+    var result: Result<CryptoAddressDetails, PayIDError> = .failure(.unknown(error: "Unknown error."))
+
+    // Use a semaphore to block and wait for the asynchronous call to complete.
+    // Capture the resolved data in result.
+    let semaphore = DispatchSemaphore(value: 0)
+    self.address(for: payID, callbackQueue: self.networkQueue) { resolvedResult in
+      // Capture the result of the call.
+      result = resolvedResult
+
+      // Signal to the semaphore to unblock the thread.
+      semaphore.signal()
+    }
+
+    // Wait for networking to complete.
+    semaphore.wait()
+
+    return result
+  }
+
+  /// Resolve the given PayID to an address.
+  ///
+  /// - Parameters:
+  ///   - payID: The PayID to resolve for an address.
+  ///   - callbackQueue: The queue to run a callback on. Defaults to the main thread.
+  ///   - completion: A closure called with the result of the operation.
   public func address(
     for payID: String,
+    callbackQueue: DispatchQueue = .main,
     completion: @escaping (Result<CryptoAddressDetails, PayIDError>) -> Void
   ) {
+    // Wrap completion calls in a closure which will dispatch to the callback queue.
+    let queueSafeCompletion: (Result<CryptoAddressDetails, PayIDError>) -> Void = {
+      completion($0)
```

Comment:
Missing use of the callbackQueue provided as a parameter to this address function. As it is now, it will always return on the calling thread.